src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -343,11 +314,36 @@ public boolean equals(Object obj) {
         for (Class<?> iface : object.getInterfaces()) {
             if (iface.getAnnotation(GraphQLTypeResolver.class) != null) {
                 builder.withInterface((GraphQLInterfaceType) getInterface(iface));
+                builder.fields(getExtensionFields(iface));
             }
         }
+
+        builder.fields(getExtensionFields(object));
+
         return builder;
     }
 
+    private List<GraphQLFieldDefinition> getExtensionFields(Class<?> object) {
+        List<GraphQLFieldDefinition> fields = new ArrayList<>();
+        if (extensionsTypeRegistry.containsKey(object)) {
+            for (Class<?> aClass : extensionsTypeRegistry.get(object)) {
+                for (Method method : getOrderedMethods(aClass)) {
+                    if (method.isBridge() || method.isSynthetic()) {
+                        continue;
+                    }
+                    if (breadthFirstSearch(method)) {
+                        GraphQLFieldDefinition field = getField(method);
+                        if (field.getDataFetcher() instanceof MethodDataFetcher) {
+                            ((MethodDataFetcher) field.getDataFetcher()).setExtension(true);
+                        }
+                        fields.add(field);
+                    }
+                }
+            }
```

Comment:
What if I put a `@GraphQLField` on a field of a class (rather than a method)?

Reply:
It is actually not supported. It was working before my last commit, but after thinking about it, I think it did not make any sense - since you are in a different object instance than the source (like a detached getter), the fields cannot be properly initialized with meaningful values. Using methods, which can use the source object passed in the constructor, makes more sense. But i could add back the parsing of the fields, if we want to support them anyway.

Reply:
I actually did not think of custom DataFetcher. I've added support and associated unit test. Thanks !